### ISSUE TITLE
wait-for-psql.py update broke my install

### ISSUE THREAD

**Issue by jwaes** [odoo/docker#278]
Since i pulled the new odoo:12 image, my odoo installation no longer starts

```
version: '2'
services:
  web:
    image: odoo:12
    depends_on:
      - db
    restart: unless-stopped
    networks:
      - odoo
      - web
    environment:
      VIRTUAL_HOST : ${HOSTNAME}
      LETSENCRYPT_HOST : ${HOSTNAME}
      LETSENCRYPT_EMAIL : jaco@example.com
      VIRTUAL_PORT : 8069
    volumes:
      - web-conf:/etc/odoo
      - web-addons:/mnt/extra-addons
      - web-lib:/var/lib/odoo
      - web-log:/var/log/odoo

  db:
    image: postgres:10
    restart: unless-stopped
    networks:
      - odoo
    environment:
      - POSTGRES_DB=postgres
      - POSTGRES_PASSWORD=odoo
      - POSTGRES_USER=odoo
    volumes:
      - db-data:/var/lib/postgresql/data

volumes:
    web-conf:
    web-addons:
    web-lib:
    web-log:
    db-data:

networks:
    odoo:
        driver: bridge
    web:
        external:
            name: web

```

i get the following error :

```
web_1  | usage: wait-for-psql.py [-h] --db_host DB_HOST --db_port DB_PORT --db_user
web_1  |                         DB_USER --db_password DB_PASSWORD [--timeout TIMEOUT]
web_1  | wait-for-psql.py: error: the following arguments are required: --db_host, --db_port, --db_user, --db_password
odoo_web_1 exited with code 2
```
Since the latest release had to do with this wait-for script i think a bug has been introduced

Starting with the previous image id hardcoded gives no problem.

**Comment by ChSch3000** [comment]
Same here. Our Docker builds with the base image odoo:12 stopped working since the wait-for update.
`FROM: odoo@sha256:dc9381042a1fa33e132ffd9d0008ab9161baaa0f72016debb947f81b467f7fa6` 
solves my problem.

**Comment by jwaes** [comment]
I reverted back to my last working image (the one from october 17th)

```
version: '2'
services:
  web:
    image: sha256:5ee30a7dd6fb5c9581af1f42fd9bfb9164bae461dbf7f5dc8a36f87a528eb21a
```

**Comment by anhvu-sg** [comment]
The same problem with my `docker-compose.yml` file. Can anyone help?

**Comment by kathpalrk** [comment]
Hi, I am having also same problem. So I've changed and revert back the changes from old Docker file to new with latest version but it's not able to work with RDS instance. I've tried to access the conatiner and there I've tried to access the RDS which is working fine but still giving an error as Internal server error. NOt sure what cause there.

**Comment by 0yik** [comment]
#279

**Comment by nextgen-networks** [comment]
> add env
> 
> * HOST=DB_HOST
> * PORT=DB_PORT
> * USER=DB_USER
> * PASSWORD=DB_PASSWORD
> 
> then if you have config file, comment db_host, db_port, db_user, db_password.
> 
> That's work for me, running docker odoo using docker-compose

Can anyone post a working docker run command example that shows the correct usage of those variables?
I was not able to find a working variant ...
Thanks!

**Comment by kovacs-andras** [comment]
> > add env
> > 
> > * HOST=DB_HOST
> > * PORT=DB_PORT
> > * USER=DB_USER
> > * PASSWORD=DB_PASSWORD
> > 
> > then if you have config file, comment db_host, db_port, db_user, db_password.
> > That's work for me, running docker odoo using docker-compose
> 
> Can anyone post a working docker run command example that shows the correct usage of those variables?
> I was not able to find a working variant ...
> Thanks!

Are you playing with the postgresql container?
A quick example from here: https://hub.docker.com/_/odoo
**$ docker run -d -e POSTGRES_USER=odoo -e POSTGRES_PASSWORD=odoo -e POSTGRES_DB=postgres --name db postgres:10**

Don't forget the PGDATA variable + volume settings also.

Check out these links: 
- https://hub.docker.com/_/postgres
- https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file
- https://vsupalov.com/docker-arg-env-variable-guide/

**Comment by kathpalrk** [comment]
its working while using the things from docker locally. If we are using the RDS database then its not working. ANy solutions for RDS on AWS-ECS cluster

**Comment by kovacs-andras** [comment]
> > > > add env
> > > > 
> > > > * HOST=DB_HOST
> > > > * PORT=DB_PORT
> > > > * USER=DB_USER
> > > > * PASSWORD=DB_PASSWORD
> > > > 
> > > > then if you have config file, comment db_host, db_port, db_user, db_password.
> > > > That's work for me, running docker odoo using docker-compose
> > > 
> > > 
> > > Can anyone post a working docker run command example that shows the correct usage of those variables?
> > > I was not able to find a working variant ...
> > > Thanks!
> > 
> > 
> > Are you playing with the postgresql container?
> > ...
> > 
> > * https://vsupalov.com/docker-arg-env-variable-guide/
> 
> Thanks @kovacs-andras but this was not what i've ment.
> I've ment env parameters used by the odoo container...
> ... as you can read in the first post these paramters are missing in the latest odoo build dated 2019-10-24.
> The link of containers seems not to work with those build.
> 
> So anyone reading here who has knowledge about overriding/force the required parameters stated as missing in first post?

Oh sorry I see now...Have you tried to set these parameters in the odoo.conf itself?
The file gets parsed from the **ODOO_RC** variable or you can bind yours to **/etc/odoo/odoo.conf**
Or you can try to set these parameters by passing them to the odoo binary itself like:
`/usr/bin/odoo  --db_host <hostname> --db_port <port> ...`

**Comment by jwaes** [comment]
The docker compose examples on https://hub.docker.com/_/odoo just do not work still ... because of these code change. This is an issue.

Also, adding environment variables for existing default values is an ugly solution.

**Comment by nextgen-networks** [comment]
> > > > > add env
...
> 
> Oh sorry I see now...Have you tried to set these parameters in the odoo.conf itself?
> The file gets parsed from the **ODOO_RC** variable or you can bind yours to **/etc/odoo/odoo.conf**
> Or you can try to set these parameters by passing them to the odoo binary itself like:
> `/usr/bin/odoo --db_host <hostname> --db_port <port> ...`

Hmm nothing seems to work really :-(
Replaced the entrypoint.sh with the latest  version from open PRs as well as validated that the odoo.conf contains the db parameters ...

Last idea of hope:
restore/pull the last working image dated 2019-10-09 of Odoo 11.
Does one of you have the digest of that version? (docker images --digests)

I've found no information how to retrieve older digests from odoo repo on docker hub.

Big thanks!

**Comment by llacroix** [comment]
@nextgen-networks if you're in a hot situation, you may want to try out my docker hub image. At least it should work. The only thing is that I often update it so it may break for a few minutes between new releases. I'm starting to roll out stable builds to prevent fetching broken odoo version and nightly alongside.

**Comment by kathpalrk** [comment]
Anyone have configured the Odoo with Lets encrypt SSL. I've tried  to use but not working. PLEASE HELP ME>

**Comment by llacroix** [comment]
@kathpalrk you're in the wrong place. The docker image shouldn't be exposed directly to the web. What you want is to setup a second service that will proxy the request to odoo. You can have a look at https://traefik.io.  It play well with docker swarm when labels are correctly set, there's little to do to make things work.

**Comment by E4eqRL0yw0yajQJ433QL** [comment]
I have exactly the same problem as described by jwaes (albeit on the odoo:11 image)
-  Is it realistic to expect that someone will pick up this issue and - at some point in time - release a new image ? 
- If not, is there anyone who can confirm that the solution offered by 0yik indeed will work (it may not be elegant, but if it works at least I have a working installation again) ?

**Comment by llacroix** [comment]
@E4eqRL0yw0yajQJ433QL 

> If not, is there anyone who can confirm that the solution offered by 0yik indeed will work (it may not be elegant, but if it works at least I have a working installation again

It should work but setting the password in an env variable shouldn't be a viable solution as it expose the password in cleartext almost everywhere. 

Passing the db_* parmeter will also get parsed as the required arguments but as explained, if you don't know the database name at boot it will fail to work too. 

You can check my image here or possibly find some other that are maintained and in a more stable release than mine. https://hub.docker.com/r/llacroix/odoo

This one may be a good alternative if you make use of oca and don't have a tool to manage images like me. https://hub.docker.com/r/elicocorp/odoo

Thought, if you insist on running the official image, here's a different solution.

If you use docker in swarm mode, replace the `command/cmd` argument to be something like `[]` or 
`""`. This will override the entrypoint and will start odoo directly skipping the whole process of parsing arguments and waiting for postgresql to boot. If you're running as a non swarm container, replace the `entrypoint` argument to `""` or `[]`.

**Comment by E4eqRL0yw0yajQJ433QL** [comment]
@0yik. Thx for posting your workaround. I can confirm that it works
@llacroix. Thx for giving me the courage to try 0yik's solution and thx for your security warning and additional possibilities
@JWaes. Thx for posting this issue in the first place. I hope that your post will (eventually) lead to an improved version of the image.

**Comment by 0yik** [comment]
you can refer to #279 to fix this problem

**Comment by nextgen-networks** [comment]
@d-fence would you please have a look to that fix?
Your latest changes lead to an unusable state of some odoo - but there seems to be a fix since a couple of days.
Thanks

**Comment by d-fence** [comment]
@nextgen-networks thanks for the ping, I' look at it

**Comment by E4eqRL0yw0yajQJ433QL** [comment]
For me release 20191106 indeed fixed the problem as mentioned in the opening post. No more tricks or workarounds needed: The new image installs smoothly. Thx @nextgen-networks for bringing this to the attention of d-fence. Thx @d-fence for picking up this up so swiftly.

**Comment by d-fence** [comment]
Fixed docker-library/official-images/commit@731144a0086baa53f681f25742143e3bbd50a800)
